benchmark/cmake/modules/AddSwiftBenchmarkSuite.cmake

```diff
@@ -267,6 +267,33 @@ function(add_swift_multisource_nonwmo_benchmark_library objfiles_out)
   set(${objfiles_out} ${objfiles} PARENT_SCOPE)
 endfunction()
 
+function(swift_benchmark_suite_version)
+    # Swift Benchmark Suite version is composed of Swift version and
+    # the git branch version.
+    message("Determining Swift Benchmark Suite version")
+    runcmd(COMMAND "${SWIFT_EXEC}" "--version"
+      VARIABLE SWIFT_VERSION
+      ERROR "Unable to determine Swift version")
+    message("Swift version: ${SWIFT_VERSION}")
+    message("CMake current source dir: ${CMAKE_CURRENT_SOURCE_DIR}")
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "describe"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    # get_current_git_branch
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "rev-parse" "--abbrev-ref" "HEAD"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    # get_git_head_ID
+    runcmd(COMMAND "git" "-C" "${CMAKE_CURRENT_SOURCE_DIR}" "rev-parse" "--short" "HEAD"
+      VARIABLE BRANCH_VERSION
+      ERROR "Unable to determine git tree version")
+    message("Git branch ${BRANCH_VERSION}")
+    configure_file(
+      ${CMAKE_CURRENT_SOURCE_DIR}/utils/Version.swift.in
+      ${CMAKE_CURRENT_SOURCE_DIR}/utils/Version.swift
```

Comment:
The pattern you cite works for generated scripts in `benchmarks/scripts/` because they are just interpreted Python. But in this case, the `Version.swift` is the input **source** file for building the benchmark binaries. I cannot just copy the source file into the binary directory,  right?  ;-)

Reply:
You are incorrect. This is the common pattern for these sorts of things. What we do is just communicate to the build system that the file to be compiled is in this other location. That is all.

Reply:
OK, I've been reading up on cmake and the whole out-of-tree build philosophy and I get your point now.